### PR TITLE
refraction law

### PR DESCRIPTION
**Pull request by paradoxx84**

### CODE REVIEW

File: symplyphysics/laws/optics/refraction_angle_from_environments.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+from symplyphysics import (
+    symbols, Eq, pretty, solve, units, Quantity, sin,
+    validate_input, validate_output, expr_to_quantity
+)
+
+from sympy.physics.units.definitions.dimension_definitions import angle as angle_type
+
+# Description
+## If ray of light comes from one media to another, it refracts.
+## Incoming ray, refracted ray and perpendicular to media boundary are within the same flat.
```
Comment (blackyblack) [inline symplyphysics/laws/optics/refraction_angle_from_environments.py:10]:
```suggestion
## Incoming ray, refracted ray and perpendicular to media boundary are within the same plane.
```

File: symplyphysics/laws/optics/refraction_angle_from_environments.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+## n1 is refractive index of first media,
+## n2 is refractive index of second media,
+## alfa is incoming angle,
+## beta is refraction angle.
+
+# Conditions
+## For the sake of sanity alfa and beta are from 0 to pi/2.
+
+refractive_index_1 = symbols('refractive_index_1')
+refractive_index_2 = symbols('refractive_index_2')
```
Comment (blackyblack) [inline symplyphysics/laws/optics/refraction_angle_from_environments.py:23]: Maybe come up with more meaningful names? I suggest `incidence_refraction_index` and `resulting_refraction_index`. For angles: `incidence_angle` and `refraction_angle`

File: symplyphysics/laws/optics/refraction_angle_from_environments.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+refraction_angle = symbols('refraction_angle')
+
+law = Eq(refractive_index_1 * sin(incoming_angle), refractive_index_2 * sin(refraction_angle))
+
+def print():
+    return pretty(law, use_unicode=False)
+
+@validate_input(index_1_=units.amount, incoming_angle_=angle_type, index_2_=units.amount)
+@validate_output(angle_type)
+def calculate_angle(index_1_: Quantity, incoming_angle_: Quantity, index_2_: Quantity) -> Quantity:
```
Comment (blackyblack) [inline symplyphysics/laws/optics/refraction_angle_from_environments.py:34]: I think refraction indexes can be plain numbers, not quantities.

Comment (blackyblack) [inline symplyphysics/laws/optics/refraction_angle_from_environments.py:34]:
```suggestion
def calculate_refraction_angle(index_1_: Quantity, incoming_angle_: Quantity, index_2_: Quantity) -> Quantity:        
```

**Review by paradoxx84 — COMMENTED** [review]
Fixes approved

**Review by blackyblack — APPROVED** [review]

**Pull request merged**